app/src/main/kotlin/com/wire/android/mapper/message/content/asset/AssetContentMapper.kt

```diff
@@ -0,0 +1,174 @@
+package com.wire.android.mapper.message.content.asset
+
+import com.wire.android.model.ImageAsset
+import com.wire.android.ui.home.conversations.model.UIMessageContent
+import com.wire.android.util.ui.WireSessionImageLoader
+import com.wire.kalium.logic.data.asset.isDisplayableImageMimeType
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.AssetContent
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.user.AssetId
+import com.wire.kalium.logic.data.user.SelfUser
+import com.wire.kalium.logic.data.user.User
+import com.wire.kalium.logic.sync.receiver.conversation.message.hasValidRemoteData
+import com.wire.kalium.logic.util.isGreaterThan
+import javax.inject.Inject
+
+class AssetContentMapper @Inject constructor(
```

Comment:
asset can be different types therefore it needed sperated mapper for indivdual types